tests/test-json-schema-to-grammar.cpp

```diff
@@ -0,0 +1,829 @@
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <regex>
+
+#include "json-schema-to-grammar.h"
+#include "grammar-parser.h"
+
+static std::string trim(const std::string & source) {
+    std::string s(source);
+    s.erase(0,s.find_first_not_of(" \n\r\t"));
+    s.erase(s.find_last_not_of(" \n\r\t")+1);
+    return std::regex_replace(s, std::regex("(^|\n)[ \t]+"), "$1");
+}
+
+enum TestCaseStatus {
+    SUCCESS,
+    FAILURE
+};
+
+struct TestCase {
+    TestCaseStatus expected_status;
+    std::string name;
+    std::string schema;
+    std::string expected_grammar;
+
+    void _print_failure_header() const {
+        std::cerr << "#" << std::endl;
+        std::cerr << "# Test '" << name.c_str() << "' failed." << std::endl;
+        std::cerr << "#" << std::endl;
+        std::cerr << schema.c_str() << std::endl;
```

Comment:
For consistency with the rest of the codebase:

`std::cerr -> fprintf(stderr,`
`std::cout -> fprintf(stdout,`

Reply:
Done